### PR TITLE
Update output adapter for validation runs

### PR DESCRIPTION
**Pull request by molly-moen**
We need to set `isValidation` on `UserTestOutputAdapter` to true on validation runs and to false on user test runs. I missed adding this in the previous work to set up validation runs. This allows us to track neighborhood requests and hide system out messages from the end user when in a validation run, and still send system out messages on user test runs.

### CODE REVIEW

**Review by jmkulwik — APPROVED** [review]
LGTM!

File: org-code-javabuilder/lib/src/main/java/org/code/javabuilder/BaseTestRunner.java [hunk 1]
```diff
@@ -22,16 +23,19 @@
...
   private final OutputAdapter outputAdapter;
   private final StatusMessageKey statusMessageKey;
+  private final boolean isValidation;
 
   public BaseTestRunner(
       List<JavaProjectFile> files,
       JavabuilderTestExecutionListener listener,
       OutputAdapter outputAdapter,
-      StatusMessageKey statusMessageKey) {
+      StatusMessageKey statusMessageKey,
```
Comment (sanchitmalhotra126) [inline org-code-javabuilder/lib/src/main/java/org/code/javabuilder/BaseTestRunner.java:32]: Could we just use the value of `statusMessageKey` to tell if this is validation or not?

Comment (molly-moen) [reply org-code-javabuilder/lib/src/main/java/org/code/javabuilder/BaseTestRunner.java:32]: good idea. I think I might actually remove `statusMessageKey` here and use `isValidation` to pick the key.

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

**Pull request merged**